### PR TITLE
Add io.github.hurricangame.hurrican

### PR DESCRIPTION
**Pull request by thrimbor**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(`device=all` for gamepads)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

### CODE REVIEW

**Comment by thrimbor** [comment]
bot, build io.github.hurricangame.hurrican

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "automerge-flathubbot-prs": true,
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:3]: :-(
any justification for restricting architectures?

File: io.github.hurricangame.hurrican.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  - name: hurrican
+    buildsystem: cmake-ninja
+    builddir: true
+    subdir: "Hurrican"
+    config-opts:
+      - "-DOPENMPT=ON"
+      - "-DCMAKE_BUILD_TYPE=Release"
+    sources:
+      - type: git
+        url: https://github.com/HurricanGame/Hurrican
```
Comment (hfiguiere) [inline io.github.hurricangame.hurrican.yml:40]: you need a specific revision

File: io.github.hurricangame.hurrican.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  - name: Icon
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 Hurrican.desktop /app/share/applications/Hurrican.desktop
```
Comment (hfiguiere) [inline io.github.hurricangame.hurrican.yml:44]: this doesn't need to be a separate module

File: flathub.json (hunk 1 above)
Comment (thrimbor) [reply flathub.json]: It was left over from the repo I used as a template. It's removed now.

File: io.github.hurricangame.hurrican.yml (hunk 2 above)
Comment (thrimbor) [reply io.github.hurricangame.hurrican.yml:42]: Done.

File: io.github.hurricangame.hurrican.yml (hunk 3 above)
Comment (thrimbor) [reply io.github.hurricangame.hurrican.yml]: Done.

File: io.github.hurricangame.hurrican.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+app-id: io.github.hurricangame.hurrican
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: hurrican
+rename-desktop-file: Hurrican.desktop
```
Comment (hfiguiere) [inline io.github.hurricangame.hurrican.yml:6]: you could just install it with the right name since the .deskop file is installed by the flatpak build

Comment (thrimbor) [reply io.github.hurricangame.hurrican.yml:6]: True, but I plan to add the .desktop file to the game repo once I've coordinated with the other dev, that's why I did it this way as preparation.

**Comment by thrimbor** [comment]
bot, build io.github.hurricangame.hurrican

File: io.github.hurricangame.hurrican.yml (hunk 4 above)
Comment (hfiguiere) [reply io.github.hurricangame.hurrican.yml:6]: It should be submitted with the proper name anyway. See https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id

File: io.github.hurricangame.hurrican.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        sha256: 50c0d62ff2d9afefa36cce9f29042cb1fb8d4f0b386b81a0fc7734f35e21e6b6
+  - name: hurrican
+    buildsystem: cmake-ninja
+    builddir: true
+    subdir: "Hurrican"
+    config-opts:
+      - "-DOPENMPT=ON"
+      - "-DCMAKE_BUILD_TYPE=Release"
+    post-install:
+      - install -Dm644 ../../Hurrican.desktop /app/share/applications/Hurrican.desktop
```
Comment (hfiguiere) [inline io.github.hurricangame.hurrican.yml:39]: you should also install the .appdata.xml

(it should also be submitted upstream)

this is why the build failed

File: io.github.hurricangame.hurrican.yml (hunk 4 above)
Comment (thrimbor) [reply io.github.hurricangame.hurrican.yml]: Ok, renamed.

File: io.github.hurricangame.hurrican.yml (hunk 5 above)
Comment (thrimbor) [reply io.github.hurricangame.hurrican.yml]: Fixed. I don't think that caused an error when building locally, is that normal?

**Comment by thrimbor** [comment]
bot, build io.github.hurricangame.hurrican

**Comment by thrimbor** [comment]
bot, build io.github.hurricangame.hurrican

File: io.github.hurricangame.hurrican.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.github.hurricangame.hurrican.metainfo.xml:1]: In general, it's better to maintain this and the desktop files in your repository, so you can update them at the same time you make a release.

**Pull request closed without merging**